test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -12888,6 +12889,9 @@ BOOST_AUTO_TEST_CASE(address_overload_resolution)
 
 BOOST_AUTO_TEST_CASE(snark)
 {
+	if (dev::test::Options::get().evmVersion() <= EVMVersion::byzantium())
```

Comment:
Can you comment on why this fails for homestead? I had a quick look at the contract below, but I didn't see why...

Reply:
(you said it probably checks if it's on homestead by checking the balance of precompiles or something like that, but I'd like to know where this happens exactly)

Reply:
It fails because the precompiles it relies on were only added in Byzantium.

Reply:
If that is the case, then why didn't the test fail before? Because we incorrectly had the precompiles for Homestead as well?

Reply:
So far I still don't really see how this test relates to the actual changes in this PR...

Reply:
Probably we should investigate more. But actually EVMHost is bad in the sense it allow the precompiles irrelevant to when they were introduced...

Reply:
If *that* is the case, then I'd expect the test case to *still* work after this change :-)... Something is weird here, is all I'm saying...

Reply:
Gitter: "The tests use call(sub(gas, 2000), … so retaining 2000 gas, but passing down the rest. Pre-byzantium no 63/64th rule existed so that was the actual gas passed down. Also “existence” rules were different. So perhaps somehow the gas calculation let it pass through (however with having a balance it just runs out of gas) and since EVMHost exposed the precompiles irrelevant to version, it just worked.
I’m pretty sure it is down to some gas calculation quirk."

Good enough for me for accepting this PR :-).

Reply:
> But actually EVMHost is bad in the sense it allow the precompiles irrelevant to when they were introduced...

Created #7868 to fix this particular issue first. Then this PR. Then evmc 7.1. And finally mocker_host. Phew, quite a bit of PRs.